File: python/345. Reverse Vowels of a String.py
#coding=utf-8
__author__ = 'xuxuan'
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        i=0
        j=len(s)-1
        a=list('aeiouAEIOU')
        s=list(s)
        while i<j:
            while i<j and s[i] not in a:
                i+=1
            while i<j and s[j] not in a:
                j-=1
            if i<j:
                temp=s[i]
                s[i]=s[j]
                s[j]=temp
                i+=1
                j-=1
        return ''.join(s)

File: java/345. Reverse Vowels of a String.java
public class Solution {
    public String reverseVowels(String s) {
        char[] cs=s.toCharArray();
        int i=0;
        int j=cs.length-1;
        while (i<j) {
			while (i<j&&cs[i]!=65&&cs[i]!=69&&cs[i]!=73&&cs[i]!=79&&cs[i]!=85&&cs[i]!=97&&cs[i]!=101&&cs[i]!=105&&cs[i]!=111&&cs[i]!=117) {
				i++;
			}
			while (i<j&&cs[j]!=65&&cs[j]!=69&&cs[j]!=73&&cs[j]!=79&&cs[j]!=85&&cs[j]!=97&&cs[j]!=101&&cs[j]!=105&&cs[j]!=111&&cs[j]!=117) {
				j--;
			}
			if (i<j) {
				char c=cs[i];
				cs[i]=cs[j];
				cs[j]=c;
				i++;
				j--;
			}
		}
        return String.valueOf(cs);
    }
}